crates/contracts/staking/src/state.rs

```diff
@@ -154,15 +154,6 @@ impl Digestable for OnChainState {
     }
 }
 
-impl Digestable for Staking {
-    fn as_digest(&self) -> sdk::StateDigest {
```

Comment:
IMHO on ne devrait pas retirer cet impl là, mais le précédent (for OnChainState), et avoir ini le as_digest qui retourne le OnChainState (as StateDigest). Je pense que la struct OnChainState ne devrait plus tellement exister